tests/test_black.py

```diff
@@ -0,0 +1,67 @@
+"""
+Contains tests for the black formatting tool integration to python_ta.check_all.
+"""
+
+import contextlib
+import io
+import subprocess
+import sys
+
+import pytest
+
+error_params = [
+    (
+        """def foo():print("Hello, world!")\nimport python_ta\npython_ta.check_all(autoformat=True)""",
+        True,
+        """def foo():\n    print("Hello, world!")\n\n\nimport python_ta\n\npython_ta.check_all(autoformat=True)\n""",
+    ),
+    (
+        """def foo():print("Hello, world!")\nimport python_ta\npython_ta.check_all(autoformat=True, config={'max-line-length': 50})""",
+        True,
+        """def foo():\n    print("Hello, world!")\n\n\nimport python_ta\n\npython_ta.check_all(\n    autoformat=True,\n    config={\'max-line-length\': 50},\n)\n""",
+    ),
+    (
+        """def foo():print("Hello, world!")\nimport python_ta\npython_ta.check_all(autoformat=False)""",
+        False,
+        """def foo():print("Hello, world!")\nimport python_ta\npython_ta.check_all(autoformat=False)""",
+    ),
+]
+
+
+@pytest.fixture
+@pytest.mark.parametrize("source", error_params)
+def unformatted_file(tmp_path, source):
+    # Create a temporary file with unformatted Python code
+    file_path = tmp_path / "sample_code.py"
+    with open(file_path, "w") as f:
+        f.write(source)
+    return file_path
+
+
+@pytest.mark.parametrize("source, pep8_check, expected_code", error_params)
+def test_black_formatting(source, pep8_check, expected_code, unformatted_file):
+    output = io.StringIO()
+
+    with contextlib.redirect_stdout(output):
+        result = subprocess.run(
```

Comment:
Overall, you shouldn't need to use `subprocess` here, and removing it may make it easier to debug the timeout issue. Call `python_ta.check_all` direction in here, and pass in the path to the temporary file.